mlflow/utils/autologging_utils.py

```diff
@@ -306,6 +310,101 @@ def batch_metrics_logger(run_id):
+    with open(version_file_path) as f:
+        return yaml.load(f, Loader=yaml.SafeLoader)
+
+
+_version_file_json = _load_version_file_json()
+
+
+# A map FLAVOR_NAME -> list of dependent (module_name, key_in_version_file_json)
+_cross_tested_flavor_to_module_name_and_json_key = {
+    "fastai": [("fastai", "fastai-1.x")],
+    "gluon": [("mxnet", "gluon")],
+    "keras": [("keras", "keras")],
+    "lightgbm": [("lightgbm", "lightgbm")],
+    "statsmodels": [("statsmodels", "statsmodels")],
+    "tensorflow": [("tensorflow", "tensorflow")],
+    "xgboost": [("xgboost", "xgboost")],
+    "sklearn": [("sklearn", "sklearn")],
+    "pytorch": [("torch", "pytorch"), ("pytorch_lightning", "pytorch-lightning")],
+}
+
+
+def _get_min_max_version_and_pip_release(json_key):
+    min_version = _version_file_json[json_key]["autologging"]["minimum"]
+    max_version = _version_file_json[json_key]["autologging"]["maximum"]
+    pip_release = _version_file_json[json_key]["package_info"]["pip_release"]
+    return min_version, max_version, pip_release
+
+
+def _is_autologging_integration_supported(flavor_name, get_module_version_fn):
+    def _check_supported(module_name, json_key):
+        min_version, max_version, _ = _get_min_max_version_and_pip_release(json_key)
+        return _check_version_in_range(get_module_version_fn(module_name), min_version, max_version)
+
+    return all(
+        [
+            _check_supported(module_name, json_key)
+            for module_name, json_key in _cross_tested_flavor_to_module_name_and_json_key[
+                flavor_name
+            ]
+        ]
+    )
+
+
+def is_autologging_integration_supported(flavor_name):
+    def get_module_version(module_name):
+        return importlib.import_module(module_name).__version__
+
+    return _is_autologging_integration_supported(flavor_name, get_module_version)
+
+
+def gen_autologging_package_version_requirements_doc(flavor_name):
+    def _gen_single_requirement(json_key):
+        min_ver, max_ver, pip_release = _get_min_max_version_and_pip_release(json_key)
+        return "``{min_ver}`` <= ``{pip_release}`` <= ``{max_ver}``".format(
+            min_ver=min_ver, pip_release=pip_release, max_ver=max_ver
+        )
+
+    required_pkg_versions = ",".join(
+        [
+            _gen_single_requirement(json_key)
+            for _, json_key in _cross_tested_flavor_to_module_name_and_json_key[flavor_name]
+        ]
+    )
+
+    return (
+        "    .. Note:: Autologging is known to be compatible with the following package versions: "
+        + required_pkg_versions
+        + ". Autologging may not succeed when used with package versions outside of this range..\n\n"
+    )
+
+
+def _log_warning_for_unsupported_integration(flavor_name):
+    _logger.warning(
+        "You are using an unsupported version of %s. If you encounter errors during autologging, try upgrading / downgrading %s to a supported version, or try upgrading MLflow.",
```

Comment:
We could wrap a long string like this to avoid a lint error:

```python
a = (
    "longlonglonglonglonglonglonglong "
    "stringstringstringstringstringstring "
)
```